### PR TITLE
add implementation for verify email for apiv4

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
Adding /users/user_id/verify/email endpoint for apiV4

#### Ticket Link
N/A

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Added API documentation (required for all new APIs) - PR: https://github.com/mattermost/mattermost-api-reference/pull/107
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

**Comment by cpanato** [comment]
Please let me know if this is ok, per conversation with @jwilander this is a copy from the existing implementation.

**Comment by cpanato** [comment]
on my local it pass as usual :)

on jenkins I'm getting

```
[2017/02/22 18:06:50 UTC] [EROR] /api/v4/users/18jjjdjpp3nozchxm4o1qpkayw/email/verify:verifyEmail code=400 rid=uc716qs6xtnstbkshpr3k7owew uid=eaqs5pgoqiy9pfwbcxaf1sw4he ip=127.0.0.1 Bad verify email link. [details: ]
goroutine 22930 [running]:
runtime/debug.Stack(0xc422b1bd08, 0xc422b1bd08, 0xc42292a980)
	/usr/local/go/src/runtime/debug/stack.go:24 +0x79
runtime/debug.PrintStack()
	/usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/mattermost/platform/api4.CheckBadRequestStatus(0xc421341980, 0xc422256180)
	github.com/mattermost/platform/api4/_test/_obj_test/apitestlib.go:502 +0x8f
github.com/mattermost/platform/api4.TestVerify(0xc421341980)
	/home/ubuntu/workspace/mattermost-platform-pr@2/src/github.com/mattermost/platform/api4/user_test.go:947 +0x380
testing.tRunner(0xc421341980, 0xc08378)
	/usr/local/go/src/testing/testing.go:610 +0x81
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:646 +0x2ec
--- FAIL: TestVerify (2.17s)
	apitestlib.go:503: actual: 301
	apitestlib.go:504: expected: 400
	apitestlib.go:505: wrong status code
```

File: model/client4.go [hunk 1]
```diff
@@ -475,6 +475,17 @@ func (c *Client4) GetAudits(userId string, page int, perPage int, etag string) (
 	}
 }
 
+// Verify user email user id and hash strings.
+func (c *Client4) Verify(userId, hashId string) (bool, *Response) {
```
Comment (jwilander) [inline model/client4.go]: Can we just update this to be `VerifyUserEmail` so it's more explicit?

**Comment by crspeller** [comment]
@jwilander Waiting on your re-review.

**Pull request merged**